Research/lva-node-app/public/js/main.js

```diff
@@ -0,0 +1,656 @@
+/**
+ * @fileoverview
+ * This file contains the majority of the functionality used across the various web pages, in the following order
+ * 1: universal functions
+ * 2: functions related to invoking methods
+ * 3: camera related functions
+ */
+
+
+//immutable global variables are written in all caps
+const PORT = 5000;
+
+/**
+ * @global variables. Mutable global variables in javascript are typically named in camelCase
+ */
+var graphTopologies = {};
+var graphInstances = {};
+var cameras = {};
+
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// universal functions
+
+/**
+* get global variables from server, Returns a promise so that functions that depend on these variables can wait
+* @returns {Promise<any>}
+*/
+function getCamerasFromServer()
+{
+    return new Promise((resolve, reject) =>
+    {
+        let request = prepareRequest(`http://localhost:${PORT}/globals`, "GET");
+        request.onreadystatechange = function ()
+        {
+            if (request.readyState == 4)
+            {
+                if (request.status == 200)
+                {
+                    let response = JSON.parse(request.response);
+                    cameras = response.cameras;
+                    resolve(response);
+                }
+                else
+                {
+                    reject("Get Globals rejection");
+                }
+            }
+            else if (request.status != 200)
```

Comment:
do we need this anymore? throughout i'm seeing all of the different implementations of onreadystatechange error handling and i'm not clear why they're all different. i think we just want to handle readyState 4 (on the assumption that always happens), and then have an if/else within there on the status.

Reply:
Yes with the new changes in place this can be taken out!